src/modules/api/transaction/services/index.ts

```diff
@@ -829,18 +826,15 @@ export class TransactionService {
 
         switch (userExists.userType) {
             case UserType.CUSTOMER: {
-                queryOptions.where.userId = +options.userId;
+                queryOptions.where.userId = id;
                 break;
             }
             case UserType.AGENT: {
-                queryOptions.where.userId = +options.userId;
+                queryOptions.where.userId = id;
                 break;
             }
             case UserType.MERCHANT: {
-                queryOptions.where.OR = [
-                    { userId: +options.userId },
-                    { merchantId: +options.userId },
-                ];
+                queryOptions.where.OR = [{ userId: id }, { merchantId: id }];
```

Comment:
you can just fetch for merchants only. we will add the feature for sub-agents later

Reply:
I've made an update to the PR